internal_transfers/actions/src/orderbook.ts

```diff
@@ -0,0 +1,69 @@
+import { WinningSettlementData } from "./models";
+import axios from "axios";
+
+// https://api.cow.fi/docs/#/
+const PROD_API_URL =
+  "https://api.cow.fi/mainnet/api/v1/solver_competition/by_tx_hash";
+const BARN_API_URL =
+  "https://barn.api.cow.fi/mainnet/api/v1/solver_competition/by_tx_hash";
+
+async function getWinningSettlementFromOrderbookAPI(
+  apiUrl: string
+): Promise<WinningSettlementData | undefined> {
+  try {
+    const response = await axios.get(apiUrl);
+
+    // The Orderbook, stores all solution submissions sorted by the objective criteria.
+    // The winning solution is the last entry of the `solutions` array.
+    const solutionArray = response.data.solutions;
+    const winningSolution = solutionArray[solutionArray.length - 1];
+    // TODO - Note that old records have solver = ZeroAddress!
+    //  We need to rectify this (use txReceipt or whatever Tenderly web-action provides)
+    const solver = winningSolution.solverAddress;
+
+    return {
+      simulationBlock: response.data.competitionSimulationBlock,
+      solver,
+      reducedCallData: winningSolution.callData,
+      fullCallData: winningSolution.uninternalizedCallData,
+    };
+  } catch (exception) {
+    if (axios.isAxiosError(exception)) {
+      if (exception.response?.status != 404) {
+        throw `ERROR response status from ${apiUrl}: ${exception.response}`;
+      }
+      // Not Found (404) is an expected response from the API.
+      // Could be for any of the following reasons:
+      // 1. !isValidTxHash (regex:  /^0x([A-Fa-f0-9]{64})$/ )
+      // 2. !isActualTxHash: web3.eth.getTransactionReceipt(txHash) -- Requires EVM Node.
+      // 3. !isSettlementTransaction(txReceipt)
+      // 4. !correctDB
```

Comment:
Not sure if it can ever happen for the usecases we are interested in, but the API is also lagging a bit behind (maybe 15-20min max) so you might not get a response if the settlement just happened.

Reply:
That is DEFINITELY an issue and very good point to raise. Will have to revisit this.

Reply:
In fact we are going to have to request that the API not be lagging behind. Do you know why that might be? If it's about block finalization then I don't believe that should affect the solver competition table since it doesn't contain mined transaction data.

Reply:
Tbh I don't remember the reason. I think the lag is 64 blocks so it might have to do with finality, but I don't remember the specifics. Maybe @sunce86 can help here.

Reply:
Ya. If it's 64 blocks then that's only related to on chain data finalization. We should be fine to fetch solver competition data as soon as it's available (before the settlement).